cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -206,12 +209,20 @@ public void deleteApp(HttpRequest request, HttpResponder responder,
                         @PathParam("namespace-id") String namespaceId,
                         @PathParam("app-id") String appId) throws Exception {
     Id.Application id = validateApplicationId(namespaceId, appId);
-    try {
-      applicationLifecycleService.removeApplication(id);
-      responder.sendStatus(HttpResponseStatus.OK);
-    } catch (CannotBeDeletedException e) {
+    Iterable<Id.Program> runningPrograms = applicationLifecycleService.removeApplication(id);
+    if (!Iterables.isEmpty(runningPrograms)) {
+      StringBuilder runningProgramsList = new StringBuilder();
+      for (Id.Program program : runningPrograms) {
+        runningProgramsList.append(program.getId());
+        runningProgramsList.append(", ");
+      }
+      String runningProgramString = runningProgramsList.toString();
```

Comment:
As a convention, don't name variables with the type. We already know that its a string, so it makes it a little awkward. However, if you address @rohitsinha54's comment, you will perhaps get rid of these local vars.